crates/re_viewport/src/viewport.rs

```diff
@@ -477,3 +452,107 @@ fn space_view_ui(
 
     space_view_blueprint.scene_ui(space_view_state, ctx, ui, latest_at, space_view_highlights);
 }
+
+struct TabWidget {
```

Comment:
very tidy, neat. I'd make it its own module but that's more of a taste thing I reckon :)